posthog/api/decide.py

```diff
@@ -244,11 +245,13 @@ def get_decide(request: HttpRequest):
 
             if feature_flags:
                 # Billing analytics for decide requests with feature flags
+                # Don't count if all requests are for survey targeting flags only.
```

Comment:
I was thinking about this problem a bit more. Can't we actually do a simplified calculation to know how many decide calls are based on surveys?

If survey has any feature flag connected to it, we can _roughly_ say its called decide at least 1x whenever it's called, right?

So wouldn't subtracting the number of decide requests that's roughly equivalent to survey API calls from the number of total decide requests they have give a closer estimate?

Let me know if my math isn't mathing here

Reply:
Surveys aren't calling flags independently.

1. posthog-js calls `/decide` to get all flags
2. Any clients that depend on flags use the return value: be it surveys, their own client, experiments, or anything else.

The math potentially works currently because surveys are used less often, but imagine a scenario like:

1. A large user enables feature flags, uses them heavily, and added one single survey as well.
2. On a page refresh we call: (1) decide to get all flags. (2) surveys API to get all surveys. (and let's say this happens a million times)
3. They call `identify()` and update user properties which refreshes the flags (a second `/decide` call).

In this case, based on current current counts we have: 2 million decide requests, 1 million survey API requests.

Going by this, we'll charge them for 1 million decide requests (2 mil - 1 mil) : which is a 50% discount on decide requests for creating a survey.

If they didn't create a survey, they'd be charged for 2 mil decide requests. 

If I were priced like this, I'd create a random survey just to get the discount. Also harder to explain + perverse incentives around faking survey API requests.

This doesn't seem reasonable to me, and in some cases (local evaluation) - the decide requests can be lower than survey requests, which leads to full discounting of decide requests 😅 .